pkg/controller/compliance/compliance_controller.go

```diff
@@ -356,6 +392,19 @@ func (r *ReconcileCompliance) Reconcile(ctx context.Context, request reconcile.R
 		return reconcile.Result{}, err
 	}
 
+	if !utils.IsFeatureActive(license, common.ComplianceFeature) {
+		log.V(4).Info("Compliance is not activated as part of this license")
+		if r.status.IsAvailable() {
+			if err := handler.Delete(context.Background(), component, r.status); err != nil {
```

Comment:
I think we already have a mechanism to delete objects that aren't needed.

Here's an example: https://github.com/tigera/operator/blob/master/pkg/render/kube-controllers.go#L156

Would it work for this to have something like this in render?

```go
func Objects() ([]client.Object, []client.Object) {
. . .

  if licenseActive {
    // Create everything
    return objs, nil
  } else {
    // Delete everything.
    return nil, objs
  }
}
```

Reply:
From what I noticed, `objectsToDelete` is nil and not populated by the renderer (for compliance and intrusion detection). I think that we should delete all the objects we are creating.

Reply:
Yes, exactly - I am suggesting that we populate that list with all of the rendered objects rather than leave it nil.

Reply:
In compliance, we are not actually respecting this contract. In MCM, we actually delete compliance server, but keep the rest of the components. I actually need to delete all created objects.

https://github.com/tigera/operator/blob/master/pkg/render/compliance.go#L200-L214.

Reply:
Update this flow as per offline discussions.